src/easylink/implementation.py

```diff
@@ -15,8 +18,16 @@ class Implementation:
     inside the container, and some metadata about the container.
     """
 
-    def __init__(self, step_name: str, implementation_config: LayeredConfigTree):
+    def __init__(
+        self,
+        step_name: str,
+        implementation_config: LayeredConfigTree,
+        input_slots: dict[str, "InputSlot"] = {},
+        output_slots: dict[str, "OutputSlot"] = {},
```

Comment:
How about if I instantiate it the same way as all the steps do, i.e.
```
    def __init__(
        self,
        name: str,
        input_slots: List[InputSlot] = [],
        output_slots: List[OutputSlot] = [],
    ):
        self.input_slots = {slot.name: slot for slot in input_slots}
        self.output_slots = {slot.name: slot for slot in output_slots}
```

Reply:
Doesn't that have the same problem, since lists are also mutable?

Reply:
I think I just didn't understand the nature of the objection, but now I do. Doesn't this suggest I should change these mutable default args *all over* step.py too?

Reply:
perhaps a compact solution for this situation is:
```
    def __init__(
        self,
        name: str,
        input_slots: Iterable[InputSlot] = (),
        output_slots: Iterable[OutputSlot] = (),
    ):
        self.input_slots = {slot.name: slot for slot in input_slots}
        self.output_slots = {slot.name: slot for slot in output_slots}
```
since we are simply iterating over the given slots in the first place. I'd prefer to avoid `None` defaults when possible for type hinting reasons.